src/core/CLucene/index/IndexWriter.cpp

```diff
@@ -1308,20 +1309,36 @@ void IndexWriter::indexCompaction(std::vector<lucene::store::Directory *> &src_d
     }
     assert(readers.size() == numIndices);
 
-    // check hasProx
+    // check hasProx, indexVersion
     bool hasProx = false;
+    IndexVersion indexVersion = IndexVersion::kV1;
     {
         if (!readers.empty()) {
+            auto release_readers = [this]() {
```

Comment:
why we release reader here?

Reply:
Release memory in the try->catch->finally code